### DISCUSSION TITLE
Addresses of traps

### DISCUSSION THREAD

**Discussion by liamappelbe** [wasmerio/wasmer#2622] [Q&A]
I'm using wasmer-c-api version 1.0.2. If I return a trap from a function created using `wasm_func_new_with_env`, and then catch the trap returned by `wasm_func_call`, are they guaranteed to have the same address, or can the wasmer runtime move the memory around or make a copy of the trap?

I've been using the raw address of the trap as a key in a map where I store extra metadata about the trap, but this doesn't seem to work reliably on windows or mac.

**Comment by Hywan** [comment]
As far as I know, the memory for traps are not moved, but it's not guaranteed. Maybe @syrusakbary can confirm that as he has worked on improving the traps API these past weeks.

**Comment by Hywan** [comment]
Let me rephrase to make it clear. The important part is: “It's not guaranteed”. We could `Pin` the trap pointers maybe, I don't know what it implies immediately, I would need time to think about it.
